File: stack.py
class Stack:
    def __init__(self):
        self.l = []

    def push(self,data):
        self.l.append(data)

    def popS(self):
        if self.size() == 0:
            raise Exception("There is no item in the stack")
        return self.l.pop()

    def size(self):
        return len(self.l)

    def peek(self):
        return self.l[self.size()-1]

if __name__ == "__main__":
    s = Stack()
    s.push(1)
    s.push(2)
    s.push(3)
    s.push(4)
    s.push(5)
    s.push(6)
    print s.size()
    print s.popS()
    print s.peek()

File: concepts/golang/stack.go
package main

import (
	"fmt"
)

// Stack represents a stack data structure.
type Stack struct {
	items []interface{}
}

// Push adds an item to the top of the stack.
func (s *Stack) Push(item interface{}) {
	s.items = append(s.items, item)
}

// Pop removes and returns the top item from the stack.
func (s *Stack) Pop() interface{} {
	if len(s.items) == 0 {
		return nil
	}
	topIndex := len(s.items) - 1
	topItem := s.items[topIndex]
	s.items = s.items[:topIndex]
	return topItem
}

// Peek returns the top item from the stack without removing it.
func (s *Stack) Peek() interface{} {
	if len(s.items) == 0 {
		return nil
	}
	return s.items[len(s.items)-1]
}

// IsEmpty returns true if the stack is empty.
func (s *Stack) IsEmpty() bool {
	return len(s.items) == 0
}

// Size returns the number of items in the stack.
func (s *Stack) Size() int {
	return len(s.items)
}

func main() {
	stack := Stack{}

	stack.Push(10)
	stack.Push(20)
	stack.Push(30)

	fmt.Println("Size:", stack.Size())
	fmt.Println("Peek:", stack.Peek())

	fmt.Println("Pop:", stack.Pop())
	fmt.Println("Size:", stack.Size())

	fmt.Println("Pop:", stack.Pop())
	fmt.Println("IsEmpty:", stack.IsEmpty())

	fmt.Println("Pop:", stack.Pop())
	fmt.Println("IsEmpty:", stack.IsEmpty())
}
